File: grind-75/cpp/242. Valid Anagram.cpp
#include <vector>
#include <iostream>
#include <string>
#include <unordered_map>
#include <unordered_set>

using namespace std;

class Solution {
public:

    unordered_map<char, int> generateCharacterFrequencyMap (string s) {
        unordered_map<char, int> charOccurrenceMap = {};

        for ( char c : s ) {
            charOccurrenceMap[c]++;
        }

        return charOccurrenceMap;
    }

    bool isAnagram(string s, string t) {

        if (s.length() != t.length()) return false;
        
        unordered_map<char, int> sCharOccurrenceMap = generateCharacterFrequencyMap(s);
        unordered_map<char, int> tCharOccurrenceMap = generateCharacterFrequencyMap(t);
        
        return sCharOccurrenceMap == tCharOccurrenceMap;
    }
};

File: grind-75/python/242. Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        if len(s) != len(t): return False
        
        s_occurrences = {}
        t_occurrences = {}

        for c in s :
            try: s_occurrences[c] = s_occurrences[c]+1
            except: s_occurrences[c] = 0
        
        for c in t :
            try: t_occurrences[c] = t_occurrences[c]+1
            except: t_occurrences[c] = 0

        return s_occurrences == t_occurrences